src/sentry/api/base.py

```diff
@@ -463,3 +465,23 @@ def track_set_commits_local(self, request: Request, organization_id=None, projec
             project_ids=project_ids,
             user_agent=request.META.get("HTTP_USER_AGENT", ""),
         )
+
+
+class _InactiveEndpoint(Endpoint):
+    def dispatch(self, request: Request, *args, **kwargs) -> HttpResponse:
+        return HttpResponse(status=status.HTTP_404_NOT_FOUND)
+
+
+def mode_limited(mode: ServerComponentMode) -> Callable[[type], type]:
+    """Decorate an endpoint class that should be active only in one mode."""
+
+    def decorator(decorated_class: type) -> type:
+        if not issubclass(decorated_class, Endpoint):
+            raise ValueError("`@mode_limited` must decorate an Endpoint class")
```

Comment:
We might also have some function views in the application that will need to be limited.

Reply:
Implemented.